pages/courses/[id]/[section]/[lesson].js

```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
     }
   }, [cohorts, user])
 
-  useEffect(async () => {
-    setLessonsSubmitted(await getLessonsSubmissions(user?.uid))
-  }, [user, open])
+  useEffect(() => {
+    const fetchLessonsSubmitted = async () => {
+      let lessonsSubmitted_ = await getLessonsSubmissions(user?.uid, cohort?.id)
+      setLessonsSubmitted(lessonsSubmitted_)
+    }
+    fetchLessonsSubmitted()
+  }, [user, cohort, open])
 
   useEffect(() => {
-    lessonsSubmitted.map((item) => {
-      if (item?.lesson === lesson && item?.user == user?.uid && item?.cohort_id == cohort?.id) {
```

Comment:
Why did you remove this check?

Reply:
In the previous code, "lessonsSubmitted" included all users' lessons from other courses and cohorts. However, my solution only fetches the lessons for the current course and cohort directly in the database query, eliminating the need for additional checks.